third_party/clangd/Protocol.cpp

```diff
@@ -0,0 +1,251 @@
+//===--- Protocol.cpp - Language Server Protocol Implementation -----------===//
```

Comment:
Are these `clangd/` files copied directly? How should they be updated? My understanding was these files were already included in our LLVM dependency. It would be better to use that instead of having two copies.